satpy/composites/__init__.py

```diff
@@ -1012,16 +1058,10 @@ def __call__(self, datasets, optional_datasets=None, **info):
                                     'the same size. Must resample first.')
 
         new_attrs = {}
-        p1, p2, p3 = datasets
         if optional_datasets:
-            high_res = optional_datasets[0]
-            low_res = datasets[["red", "green", "blue"].index(
-                self.high_resolution_band)]
-            if high_res.attrs["area"] != low_res.attrs["area"]:
-                    raise IncompatibleAreas("High resolution band is not "
-                                            "mapped to the same area as the "
-                                            "low resolution bands. Must "
-                                            "resample first.")
+            datasets = self.check_areas(datasets + optional_datasets)
```

Comment:
some compositors (palettecompositor) have a palette as optional dependency, and the palette has no area attribute, and has a different shape.

Reply:
So maybe check that the dimensions are called `x` and `y` before going further ? (although I know we're not there yet)

Reply:
Checking dimensions sounds good and makes sense. However this specific line is just for ratio sharpening so it should never be a palette.